pkg/collections/transientdata/storeprovider/store/dbstore/dbstore.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package dbstore
+
+import (
+	"bytes"
+	"encoding/gob"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hyperledger/fabric/common/flogging"
+	"github.com/hyperledger/fabric/common/ledger/util/leveldbhelper"
+	"github.com/pkg/errors"
+	"github.com/trustbloc/fabric-peer-ext/pkg/collections/transientdata/storeprovider/store/api"
+)
+
+var logger = flogging.MustGetLogger("transientdb")
+
+var compositeKeySep = "!"
+
+// DBStore holds the db handle and the db name
+type DBStore struct {
+	db     *leveldbhelper.DBHandle
+	dbName string
+}
+
+// newDBStore constructs an instance of db store
+func newDBStore(db *leveldbhelper.DBHandle, dbName string) *DBStore {
+	return &DBStore{db, dbName}
+}
+
+// AddKey add cache key to db
+func (s *DBStore) AddKey(key api.Key, value *api.Value) error {
+	encodeVal, err := encodeCacheVal(value)
+	if err != nil {
+		return errors.WithMessagef(err, "failed to encode transientdata value %s", value)
+	}
+	// put key in db
+	err = s.db.Put(encodeCacheKey(key, time.Time{}), encodeVal, true)
+	if err != nil {
+		return errors.Wrapf(err, "failed to save transientdata key %s in db", key)
+	}
+
+	if !value.ExpiryTime.IsZero() {
+		// put same previous key with prefix expiryTime so the clean up can remove all expired keys
+		err = s.db.Put(encodeCacheKey(key, value.ExpiryTime), []byte(""), true)
+		if err != nil {
+			return errors.Wrapf(err, "failed to save transientdata key %s in db", key)
+		}
+	}
+
+	return nil
+}
+
+// GetKey get cache key from db
+func (s *DBStore) GetKey(key api.Key) (*api.Value, error) {
+	logger.Debugf("load transientdata key %s from db", key)
```

Comment:
should %s be replaced with %v? or maybe api.Key can be output as string?

Reply:
Key has a String() function.